src/main/java/com/coreos/jetcd/MaintenanceImpl.java

```diff
@@ -38,10 +43,34 @@
 
   private MaintenanceGrpc.MaintenanceFutureStub futureStub;
   private MaintenanceGrpc.MaintenanceStub streamStub;
+  private DialFunction dialFunction;
+  private ExecutorService executorService;
 
-  public MaintenanceImpl(ManagedChannel channel, Optional<String> token) {
-    this.futureStub = ClientUtil.configureStub(MaintenanceGrpc.newFutureStub(channel), token);
-    this.streamStub = ClientUtil.configureStub(MaintenanceGrpc.newStub(channel), token);
+  public MaintenanceImpl(Client c) {
+    dialFunction = (endpoint) -> {
+      NameResolver.Factory nameResolverFactory = ClientUtil
+          .simpleNameResolveFactory(Arrays.asList(endpoint));
+      return c.toChannelAndToken(defaultChannelBuilder(nameResolverFactory));
+    };
+    
+    this.executorService = c.getExecutorService();
+
+    this.futureStub = ClientUtil
+        .configureStub(MaintenanceGrpc.newFutureStub(c.getChannel()), c.getToken());
+    this.streamStub = ClientUtil
+        .configureStub(MaintenanceGrpc.newStub(c.getChannel()), c.getToken());
+  }
+
+  @FunctionalInterface
+  private interface DialFunction {
+
+    /**
+     * dial dials to an endpoint and returns a managed channel its associated token.
+     *
+     * @return a managed channel its associated token.
```

Comment:
the token here is same as the one in go client counter part at https://github.com/coreos/etcd/blob/master/clientv3/client.go#L286.  If auth is enabled for etcd, the this token is needed to build a stub that has the token as its metadata. then this token is passed to etcd server for authentication.